### PR TITLE
[add] post unlike

### PR DESCRIPTION
**Pull request by SapeleD3**

### CODE REVIEW

**Review by EngrEric — CHANGES_REQUESTED** [review]
remove the `/* eslint-disable indent */` and rename the var name `file` to be more verbose

File: functions/controllers/post/unlikePost.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/* eslint-disable indent */
+/* eslint-disable no-tabs */
```
Comment (Way2nnadi) [inline functions/controllers/post/unlikePost.js:2]: do not disable the eslint

File: functions/controllers/post/unlikePost.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const { error, success } = status;
+const { postNotLiked, somethingWentWrong, postNotFound } = message;
+
+let postData;
+
+const unlikePosts = async (req, res, db, postDoc, file) => {
+	await db.doc(`/likes/${file.docs[0].id}`).delete();
+	// eslint-disable-next-line no-plusplus
+	postData.likeCount--;
```
Comment (Way2nnadi) [inline functions/controllers/post/unlikePost.js:14]: Do not disable this line. Be explicit with your code `postData.likeCount -= 1;`

File: functions/controllers/post/unlikePost.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			.status(BAD_REQUEST)
+			.json({ message: postNotFound, status: error });
+	}
+	if (file.empty) {
+		return res
+			.status(BAD_REQUEST)
+			.json({ message: postNotLiked, status: error });
+	}
+	return unlikePosts(req, res, db, postDoc, file);
+};
```
Comment (Way2nnadi) [inline functions/controllers/post/unlikePost.js:36]: The flow of this function is weird. Try something like this
```
const file = await likeDoc.get();

if (!post.exists || file.empty) {
  return res
    .status(BAD_REQUEST)
    .json({ message: postNotFound, status: error });
} 

postData = post.data();
postData.postId = post.id;

return unlikePosts(req, res, db, postDoc, file);
```

Comment (Way2nnadi) [reply functions/controllers/post/unlikePost.js:36]: wait where is `postData` defined?

**Review by Way2nnadi — CHANGES_REQUESTED** [review]
You need to fix that function logic

File: functions/controllers/topic/getTopic.js [hunk 4]
```diff
@@ -1,42 +1,43 @@
-const _ = require('lodash');
-const HttpStatus = require('http-status-codes');
+/* eslint-disable arrow-parens */
+/* eslint-disable indent */
+/* eslint-disable no-tabs */
```
Comment (Way2nnadi) [inline functions/controllers/topic/getTopic.js]: remove this disables man

File: functions/controllers/topic/getTopic.js [hunk 5]
```diff
@@ -1,42 +1,43 @@
...
+		return res
+			.status(INTERNAL_SERVER_ERROR)
+			.json({ message: somethingWentWrong, status: error });
+	}
 };
 
 module.exports = {
-    getAllTopics,
+	getAllTopics,
 };
```
Comment (Way2nnadi) [inline functions/controllers/topic/getTopic.js]: all the indents on this file is wrong

File: functions/controllers/topic/index.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable indent */
+/* eslint-disable no-tabs */
```
Comment (Way2nnadi) [inline functions/controllers/topic/index.js]: remove the disables

**Pull request merged**